test/test_other.py

```diff
@@ -400,6 +400,18 @@ def test_export_es6_allows_export_in_post_js(self):
     src = read_file('a.out.js')
     self.assertContained('export{doNothing};', src)
 
+  def test_export_es6_pthread(self):
+    # Test ES6 + pthreads + running the output in node.
+    self.run_process([EMCC, test_file('hello_world.c'), '-sEXPORT_ES6',
+                      '-pthread', '-o', 'hello.mjs'])
+    # Node requires that we load the module in the following manner (this is
+    # parallel to the code we emit for HTML in ScriptSource.replacement).
+    create_file('runner.mjs', '''
+      import initModule from "./hello.mjs";
+      initModule();
```

Comment:
I am unsure why this `initModule` pattern is needed. It is already in use in the codebase as the comment mentions, and it works, so it looks correct. But oddly I can't find an `initModule` method anywhere in our codebase or in the node or JS docs..?

@sbc100 @RReverser maybe you know?

Reply:
Oh, I guess this is just executing the default export from the module? What we used to call "MODULARIZE_INSTANCE"...

Reply:
The default export is called `Module`.. not `initModule` and it should return a promise of a module so you would need to await on it.

Reply:
@kripken is correct. The actual name you bind the default export to doesn't matter / doesn't have to match between import and export sides.

Reply:
We should perhaps unify this method across all JS module testing we do here in `test_core.py` or `test_other.py`.

How about maybe a new `self.run_js_module('hello.mjs')` that know how to run emscripten-built js modules?  (i.e. takes are of building the runner.mjs).

Reply:
@sbc100 Do we test this in core? All I found is testing in the browser, and some compilation-only tests in other. This is the first non-browser test to actually run `EXPORT_ES6` code AFAICT.

With that said I like the idea to add `run_js_module` but it would only be used in this one function for now?

Reply:
If this is the only place we test `EXPORT_ES6` so far in `test_core.py` then maybe just leave it as is and we can refactor later..

I'm a little surprised that it is!

Reply:
Yeah, I was surprised too...

My guess is that we couldn't add ES6 testing when our node version was too old. Then we updated node but didn't add testing. But now we can :rocket:

Reply:
In that case maybe just call this `test_export_es6` and have a `pthread` variant of it?

Reply:
Good idea, done.